osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/securitygroup/CheckK8sSecurityGroupLabelMetaTask.java

```diff
@@ -18,39 +18,65 @@
 
 import javax.persistence.EntityManager;
 
-import org.apache.commons.lang.NotImplementedException;
-import org.osc.core.broker.job.Task;
+import org.apache.log4j.Logger;
 import org.osc.core.broker.job.TaskGraph;
 import org.osc.core.broker.model.entities.virtualization.SecurityGroupMember;
 import org.osc.core.broker.service.tasks.TransactionalMetaTask;
 import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
 
 @Component(service = CheckK8sSecurityGroupLabelMetaTask.class)
 public class CheckK8sSecurityGroupLabelMetaTask extends TransactionalMetaTask {
 
+    private static final Logger LOG = Logger.getLogger(CheckK8sSecurityGroupLabelMetaTask.class);
+
+    private TaskGraph tg;
     private SecurityGroupMember sgm;
     private boolean isDelete;
 
+    @Reference
+    UpdateK8sSecurityGroupMemberLabelMetaTask updateK8sSecurityGroupMemberLabelMetaTask;
+
+    @Reference
+    MarkSecurityGroupMemberDeleteTask markSecurityGroupMemberDeleteTask;
+
+    @Reference
+    SecurityGroupMemberDeleteTask securityGroupMemberDeleteTask;
+
     @Override
     public TaskGraph getTaskGraph() {
-        // TODO unimplemented
-        throw new NotImplementedException(getClass());
+        return this.tg;
     }
 
     @Override
     public String getName() {
-        return "Check Kubernetes Security Group Label Meta Task -- unimplemented " + this.sgm.getId();
+        return "Check Kubernetes Security Group Label Meta Task for security group member " + this.sgm.getId()
```

Comment:
neat!

Reply:
avoid using "Meta task" in the task name. Change to String.format("Check Security Group Label '%s' ", this.sgm.getLabel().getName());

Also use string.format for messages with multiple variables.

If isDelete true change to message to String.format("Deleting Security Group Label '%s' ", sgm.getLabel().getName());

Reply:
Here it's only  one variable